### PR TITLE
16494 Protractor (JavaScript e2e) tests

### PR DESCRIPTION
**Pull request by myabc**
:warning: **Includes commits from #2067. Please review, merge first.**

https://www.openproject.org/work_packages/16494

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://feature-16494-protractor-tests.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

File: app/assets/javascripts/angular/work_packages/tabs/index.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  ])
+  .directive('userField', ['PathHelper', require('./user-field-directive')])
+  .directive('workPackageRelations', [
+    'I18n',
+    'WorkPackagesHelper',
+    '$timeout',
+    require('./work-package-relations-directive')
+  ]);
+
+// FIXME: move modules or files to the right place
```
Comment (myabc) [inline app/assets/javascripts/angular/work_packages/tabs/index.js:60]: @hschink any idea why these directives were placed in a different module?

Comment (hschink) [inline app/assets/javascripts/angular/work_packages/tabs/index.js:60]: Sorry, I have no idea

File: .travis.yml [hunk 2]
```diff
@@ -30,7 +30,9 @@ language: ruby
 rvm:
   - 2.1
 sudo: false
-cache: bundler
+cache:
+  - bundler
+  - npm
```
Comment (myabc) [inline .travis.yml:35]: N.B. I don't believe this is working yet.

**Comment by myabc** [comment]
When reviewing, please bear in mind updated INSTALL and RUNNING_TESTS docs (see d1b778d and da316b7).

**Comment by myabc** [comment]
@0xF013 please take a look!

**Comment by 0xF013** [comment]
@myabc I am having some troubles starting protractor

```
$ gulp tests:protractor
[11:26:36] Using gulpfile ~/Work/openproject/gulpfile.js
[11:26:36] Starting 'webdriver:update'...
[11:26:36] Starting 'webpack'...
[11:26:37] Starting 'sass'...
[11:26:37] Starting 'express'...
selenium standalone is up to date.
chromedriver is up to date.
[11:26:37] Finished 'express' after 114 ms
Listening at localhost:8080
[11:26:37] Finished 'webdriver:update' after 154 ms
[11:26:38] gulp-ruby-sass: directory
[11:26:38] Version: webpack 1.4.9
                    Asset     Size  Chunks             Chunk Names
openproject-app.bundle.js  1941589       0  [emitted]  main
[11:26:38] Finished 'webpack' after 1.23 s
[11:26:38] gulp-ruby-sass: write default.css.css
[11:26:38] Finished 'sass' after 1.96 s
[11:26:38] Starting 'tests:protractor'...
Starting selenium standalone server...

/Users/elvis/Work/openproject/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:1745
      throw error;
            ^
Error: Timed out waiting for the WebDriver server at http://10.1.2.32:51517/wd/hub
    at Error (<anonymous>)
    at onResponse (/Users/elvis/Work/openproject/node_modules/protractor/node_modules/selenium-webdriver/http/util.js:87:11)
    at /Users/elvis/Work/openproject/node_modules/protractor/node_modules/selenium-webdriver/http/util.js:42:21
    at /Users/elvis/Work/openproject/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/http/http.js:96:5
    at ClientRequest.<anonymous> (/Users/elvis/Work/openproject/node_modules/protractor/node_modules/selenium-webdriver/http/index.js:145:7)
    at ClientRequest.EventEmitter.emit (events.js:95:17)
    at Socket.socketErrorListener (http.js:1547:9)
    at Socket.EventEmitter.emit (events.js:95:17)
    at net.js:440:14
    at process._tickCallback (node.js:415:13)

/Users/elvis/Work/openproject/gulpfile.js:86
      throw e;
            ^
Error in plugin 'gulp-protractor'
protractor exited with code 8
    at ChildProcess.<anonymous> (/Users/elvis/Work/openproject/node_modules/gulp-protractor/index.js:59:27)
    at ChildProcess.EventEmitter.emit (events.js:98:17)
    at Process.ChildProcess._handle.onexit (child_process.js:797:12)
```

**Comment by 0xF013** [comment]
@myabc will we have code minification?

**Comment by myabc** [comment]
@0xF013 yes, but we first have to a) either audit our codebase and make sure we use explicit annotations (Upgrading to Angular 1.3. would [help with this](https://docs.angularjs.org/error/$injector/strictdi)) OR b) incorporate a plugin like ng-annotate to annotate explicit dependencies before minification.

**Pull request merged**